### PR TITLE
Add: per-use customisable cargo list separators

### PR DESCRIPTION
**Pull request by rubidium42**
## Motivation / Problem

#11379 removed the and/or in the translation for: "Station now accepts X and Y" or "Station no longer accepts X or Y", and replaced it with commas. This PR aims to reintroduce the possibility to use 'or' or 'and' by making the separator of the `{CARGO_LIST}` command configurable by the translator.

## Description

Changes `{CARGO_LIST}` to `{CARGO_LIST last_separator normal_separator}`. In this last_separator and normal_separator are by default ", " and can be overridden. If you want to override the normal_separator, you must also set the last_separator.

When building a list of cargoes, the last separator will be used between the last two cargoes. The normal separator will be used between any earlier pairs. So practically `Cargo 1 normal separator Cargo 2 normal separator Cargo 3 last separator Cargo 4`. When there is only one cargo, no separator is used.

## Limitations

Not sure whether this changes affects the translator in any way or form. I also do not have an idea how to fix that, so if that's needed any help would be greatly appreciated.
The linguistics of when to use 'and' or 'or' in the strings might not be 100% right. It depends a bit on the actual meaning.

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, game_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

File: src/lang/english.txt [hunk 1]
```diff
@@ -924,8 +924,8 @@ STR_NEWS_NEW_VEHICLE_NOW_AVAILABLE_WITH_TYPE                    :{BLACK}New {STR
 
 STR_NEWS_SHOW_VEHICLE_GROUP_TOOLTIP                             :{BLACK}Open the group window focused on the vehicle's group
 
-STR_NEWS_STATION_NO_LONGER_ACCEPTS_CARGO_LIST                   :{WHITE}{STATION} no longer accepts: {CARGO_LIST}
-STR_NEWS_STATION_NOW_ACCEPTS_CARGO_LIST                         :{WHITE}{STATION} now accepts: {CARGO_LIST}
+STR_NEWS_STATION_NO_LONGER_ACCEPTS_CARGO_LIST                   :{WHITE}{STATION} no longer accepts: {CARGO_LIST " or "}
+STR_NEWS_STATION_NOW_ACCEPTS_CARGO_LIST                         :{WHITE}{STATION} now accepts: {CARGO_LIST " and "}
```
Comment (PeterN) [inline src/lang/english.txt:928]: In English this colon should be removed as the list is now part of a full sentence.

File: src/strgen/strgen_base.cpp [hunk 2]
```diff
@@ -419,6 +422,24 @@ void EmitGender(Buffer *buffer, char *buf, int)
...
+	if (word != nullptr) {
+		last_separator = word;
+		word = ParseWord(&buf);
+		if (word != nullptr) separator = word;
+	}
+
+	buffer->AppendUtf8(SCC_CARGO_LIST);
+	EmitWord(buffer, separator);
+	EmitWord(buffer, last_separator);
+}
```
Comment (PeterN) [inline src/strgen/strgen_base.cpp:441]: As an English-pedant, I would like Oxford-comma support:

1; Passengers
2: Passengers and Mail
3: Passengers, Mail, and Goods

There may be other rules in other languages too though.

Comment (rubidium42) [reply src/strgen/strgen_base.cpp:441]: Though shouldn't the newspaper headlines/texts follow the style guides of British newspapers?

The Times: "Avoid the so-called Oxford comma" (https://web.archive.org/web/20080724151906/http://www.timesonline.co.uk/tol/tools_and_services/specials/style_guide/article986734.ece)
The Guardian: "a comma before the final “and” in lists: straightforward ones [...] do not need one" (https://www.theguardian.com/guardian-observer-style-guide-o)
Fun thing: some newspapers/headlines occasionally omit the "and" completely in favour of a comma to shorten texts.

Not to mention that "The Oxford Style Manual" requires the Oxford-comma, whereas the "University of Oxford Public Affairs Directorate Writing and Style Guide" generally dislikes the Oxford-comma unless it reduces ambiguity.

**Comment by rubidium42** [comment]
Getting this into a releasable state, i.e. with proper translator support and all/every combination of punctuation that one can think of depending on number of items, is not something I'm going to be spending time on.
It seemed small and simple, but it seems to be spiralling out of control.

**Pull request closed without merging**